RecoBTag/FeatureTools/src/deep_helpers.cc

```diff
@@ -148,6 +148,30 @@ namespace btagbtvdeep {
     return target_length;
   }
 
+  int center_norm_pad_halfRagged(const std::vector<float> &input,
+                      float center,
+                      float norm_factor,
+                      unsigned target_length,
+                      std::vector<float> &datavec,
+                      int startval,
+                      float pad_value,
+                      float replace_inf_value,
+                      float min,
+                      float max) {
+    // do variable shifting/scaling/padding/clipping in one go
+
+    assert(min <= pad_value && pad_value <= max);
+
+    for (unsigned i = 0; i < target_length; ++i) {
```

Comment:
this loop only needs to run up to `input.size()`. then the pad value can be filled for the remainder (`target_length-input.size()` using `insert()`. that removes the if/else executed for every value.

Reply:
I see what you mean, but now I'm actually a little confused. `TritonInputData::allocate()` only calls `reserve()` on the inner vector, not `resize()`. Therefore, you should be using `push_back()` in `center_norm_pad_halfRagged()`, not just assigning. Then, `insert()` should work fine to replace the second loop.

Reply:
Ahh, huh, I really have verified that what I have now gives the same results as the non-sonic version.  I'll check and see if it also work with push_back

Reply:
Yes, I think assigning after `reserve()` technically works, but it's rather precarious (e.g. `vector::size()` won't report the right value) and not advisable.

Reply:
Fixed in the lastest push.  Uses insert now too